### ISSUE TITLE
[BUG] Edited video not loaded correctly (2.0.0-dev.2 and 1.3.10)

### ISSUE THREAD

**Issue by Tom3652** [fluttercandies/flutter_photo_manager#678]
**Describe the bug**

Initially i had the following issue with ```photo_manager: ^1.3.10``` : 
- If you trim / resize a video with your iPhone’s photo app (official from Apple), they will ask “replace existing video” or “save as new file”.
- If you choose “replace existing video”, they always keep the original file somewhere anyway so you can revert changes later.

***The problem :***

- If you resize your video (lower ```width```) for example, ```photo_manager``` ```AssetEntity``` class will always return the original width / height and not the cropped result.

This result as black screen that has as size : ```originalWidth - resizedWidth```  on edited videos when loaded from ```photo_manager```.

With ```photo_manager: ^2.0.0-dev.2``` : 

- The edited file seems to never be retrieved.  
- You can also try to crop / trim the same way as above, and you will see the original file fetched anyway using any method of ```AssetEntity``` (i have tried them all, even with ```isOrigin: false```), not the edited one.

**To Reproduce**

1. Open the ```photos``` app in you iPhone
2. Select a Video
3. Edit the video and crop it (width)
4. Replace the existing video

**Expected behavior**

I expect the new edited video to be retrieved with the (new) correct width / height rendered after cropping.

**Screenshots**

https://user-images.githubusercontent.com/32631467/148651221-2897dea8-7f0f-43b9-afbe-b10e5ec5c6f1.mov

The above was made with ```photo_manager: ^1.3.10```. The initial video ratio was indeed this one including the black part.  
With the new version, with this same video i am having the original one, not cropped or anything. 

**Flutter version**

```
[✓] Flutter (Channel stable, 2.8.1, on macOS 12.1 21C52 darwin-x64, locale fr-FR)
    • Flutter version 2.8.1 at /Users/foxtom/Desktop/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 77d935af4d (3 weeks ago), 2021-12-16 08:37:33 -0800
    • Engine revision 890a5fca2e
    • Dart version 2.15.1

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/foxtom/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.2.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • CocoaPods version 1.11.2

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] VS Code (version 1.63.2)
    • VS Code at /Users/foxtom/Desktop/Visual Studio Code.app/Contents
    • Flutter extension version 3.32.0

[✓] Connected device (3 available)
    • Now You See Me (mobile) • 00008020-001204401E78002E • ios            • iOS 15.2 19C56
    • macOS (desktop)         • macos                     • darwin-x64     • macOS 12.1 21C52 darwin-x64
    • Chrome (web)            • chrome                    • web-javascript • Google Chrome 97.0.4692.71

• No issues found!
```

**Smartphone (please complete the following information):**

 - Device: iPhone XS Max, iPhone 12 Prox Max, iPhone 11, iPhone 12
 - OS: iOS 14+ and iOS 15+

**Comment by AlexV525** [comment]
I'm not able to reproduce this issue on the latest commit.

Full operation recording:

https://user-images.githubusercontent.com/15884415/149732180-cca2e91e-d611-4720-971a-e84bdb64679d.mp4

**Comment by AlexV525** [comment]
I'm still unable to reproduce this on the latest commit with the `file` getter. Closing as it's not reproducible.

https://user-images.githubusercontent.com/15884415/149878728-cd7288a6-26e7-4da4-976e-2b6d022b706f.mp4

**Comment by Tom3652** [comment]
Hello @AlexV525, i am attaching the below video so you see the result on a real device, because it seems to work only on iOS simulators. 
Here it's tested on iPhone 8, but this happens with all of my real devices (7 different iPhones and 2 iPads). 

https://drive.google.com/file/d/123Genk6GVOYiFo1L2ArXVk_TmoS9kvKj/view?usp=sharing

You will notice both of the issues i have made in the above video : 

- #679 : The first time we call ```entity.file```, this freeze the app during few seconds and the progress is not emitted, the state success is done after the freeze is gone (10-15 sec for larger files even on very fast devices such as iPhone 13 Pro max - XS max...) 
So you were right, the file is ```locallyAvailable``` but so laggy that i thought it was coming from iCloud or downloaded anyway.  
In my app UI, i am cheating a bit to have a fluent behavior, so i have not this lag but the video usually starts only after 15 secs in average and the file is not available during all this time.
I agree with you to close #679, the ```PMProgressHandler``` has no issue as it should not be emitted in that case but it's not normal that it takes that long for a ```locallyAvailable``` file and that's the behavior my users are complaining about.    
I will create a new issue for this to track different things as you asked.

- The edited video is not loaded correctly with ```entity.file```.  
During the video recording i have made i make a full app reload to make sure the ```entity.file``` is indeed the latest, edited one.

Here is the code i have changed in your example app to reproduce it : 

```
import 'dart:io';

import 'package:flutter/material.dart';
import 'package:photo_manager/photo_manager.dart';
import 'package:video_player/video_player.dart';

class VideoWidget extends StatefulWidget {
  const VideoWidget({
    Key? key,
    required this.entity,
    this.isAudio = false,
  }) : super(key: key);

  final AssetEntity entity;
  final bool isAudio;

  @override
  _VideoWidgetState createState() => _VideoWidgetState();
}

class _VideoWidgetState extends State<VideoWidget> {

  VideoPlayerController? _controller;

  @override
  void initState() {
    super.initState();
    widget.entity.file.then((File? file) {
      if (!mounted || file == null) {
        return;
      }
      _controller = VideoPlayerController.file(file)..initialize()..addListener(() => setState(() {}));
      setState(() {

      });
    });
  }

  @override
  void dispose() {
    if (_controller != null) {
      _controller!.dispose();
    }
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return _controller != null && _controller!.value.isInitialized
        ? AspectRatio(
            aspectRatio: widget.isAudio ? 1 : _controller!.value.aspectRatio,
            child: GestureDetector(
              child: buildVideoPlayer(),
              onTap: () {
                _controller!.value.isPlaying
                    ? _controller!.pause()
                    : _controller!.play();
                setState(() {});
              },
            ),
          )
        : Container();
  }

  Stack buildVideoPlayer() {
    Widget contentWidget;
    if (!widget.isAudio) {
      contentWidget = VideoPlayer(_controller!);
    } else {
      contentWidget = Container(
        color: Colors.white,
        child: const Center(
          child: Icon(
            Icons.audiotrack,
            size: 200,
            color: Colors.grey,
          ),
        ),
      );
    }

    final List<Widget> children = <Widget>[contentWidget];

    if (!_controller!.value.isPlaying) {
      children.add(
        IgnorePointer(
          child: Center(
              child: Container(
            padding: const EdgeInsets.all(15),
            decoration: BoxDecoration(
              color: Colors.black38,
              borderRadius: BorderRadius.circular(30),
            ),
            child: const Icon(
              Icons.play_arrow,
              color: Colors.white,
            ),
          )),
        ),
      );
    }

    return Stack(
      children: children,
    );
  }
}
```



```
Widget buildVideo() {
    if (widget.mediaUrl == null) {
      return const SizedBox.shrink();
    }
    return VideoWidget(
      isAudio: widget.entity.type == AssetType.audio,
      entity: widget.entity,
    );
  }
```

**Comment by Tom3652** [comment]
Here is the render of this same cropped video in my real app : 

https://user-images.githubusercontent.com/32631467/149954417-97afb104-02f0-4f4e-a7c0-8098d54a6c2a.mov

**Comment by AlexV525** [comment]
Did you already run this on the latest commit (a8ffd5d63017059c7543438c2bc6c0f2d2c5ad46)?

**Comment by AlexV525** [comment]
Good, I'm now able to reproduce this.

**Comment by Tom3652** [comment]
Thanks for your work on this !
However, i have tested it again with the `flutter_photo_manager` latest code (`2.0.0-dev.3`) and i have now the same preview as in my real app, meaning with the black part of the screen : 

https://user-images.githubusercontent.com/32631467/150109912-dcc72ee2-1e17-469b-bf9f-ae6cdb4914cb.MOV

**Comment by AlexV525** [comment]
This is not the problem with our plugin AFAICT.

**Comment by AlexV525** [comment]
Please file new issues to track the blank area, if it's reproducible.
